tests/test_fixtures.py

```diff
@@ -32,19 +32,44 @@ def test_workspace_instance_scope_part2(workspace):
     assert workspace is workspace_instance[0]
 
 
-def test_hardware_node_add_remove(hardware):
+def _hardware_add_node(h, name, role) -> NodeBase:
```

Comment:
(nit) We might want to move this into a common lib in the future.

Reply:
yes